### DISCUSSION TITLE
How to represent ISO 8601 Dates?

### DISCUSSION THREAD

**Discussion by TonyValenti** [dotnet/runtime#64797] [Q&A] [answered]
Hi All,
The DateTime and DateTimeOffset classes have a min value of 0001-01-01.
I need to represent ISO 8601 dates which includes date=0000-01-01.

(Yes, we're actually dealing with issues because of year zero in client data).

Does anyone have any suggestions for dealing with this?

**Comment by huoyaoyuan** [comment]
Well, if the value range is clear, and the format is fixed, I think special-casing should be easiest. You can transfer it in a custom struct, and borrow regular `DateTime`/`DateOnly` operations when it does not fall into the special range.
Or you can reject it earlier to prevent polluting the business code.

> because of year zero in client data

It shouldn't be correct data, so it's some sort of fixing corruption.

**Reply by TonyValenti** [reply]
@huoyaoyuan - Except 0000-01-01 is a valid ISO 8601 date.

**Reply by huoyaoyuan** [reply]
I mean it shouldn't be a realistic date in your business logic.

**Reply by Clockwork-Muse** [reply]
@huoyaoyuan - year-0 dates are used for astronomy calculations, possibly other areas.  Although if that was the reason, I'd expect the lack of BC/BCE/negative years to also be a problem.

**Comment by Clockwork-Muse** [comment, marked as answer]
@TonyValenti - 
For better date/time in C#, use [NodaTime](https://nodatime.org/).
